### PR TITLE
Updated C++ style

### PR DESCRIPTION
**Pull request by peter-b**
Update C++ features and style guide documentation to reflect C++ coding style meeting.

### CODE REVIEW

File: docs/development/C++-features.md [hunk 1]
```diff
@@ -35,37 +35,47 @@ than compiler support.
...
 > considered before being accepted.
 
-#### `dynamic_cast` and RTTI (`typeid`, `typeinfo`): PROHIBITED
+#### `dynamic_cast` and RTTI (`typeid`, `typeinfo`): PARTIALLY PROHIBITED
+
+Use `dynamic_cast` when casting within a type hierarchy.  This ensures
+early error detection when an instance is cast to an incompatible
+type.
+
+Other uses or RTTI is currently prohibited.
```
Comment (runrevmark) [inline docs/development/C++-features.md]: or = of

**Comment by peter-b** [comment]
@runrevmark Thanks, I've fixed that typo.

File: docs/development/C++-features.md [hunk 2]
```diff
@@ -35,37 +35,47 @@ than compiler support.
...
 > considered before being accepted.
 
-#### `dynamic_cast` and RTTI (`typeid`, `typeinfo`): PROHIBITED
+#### `dynamic_cast` and RTTI (`typeid`, `typeinfo`): PARTIALLY PROHIBITED
+
+Use `dynamic_cast` when casting within a type hierarchy.  This ensures
+early error detection when an instance is cast to an incompatible
+type.
+
+Other uses of RTTI is currently prohibited.
```
Comment (livecodeali) [inline docs/development/C++-features.md]: is -> are

Comment (livecodeali) [reply docs/development/C++-features.md]: or uses -> use I guess :-)

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 82309bd

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 82309bddd853725a290eee631954fad97cad050e

**Comment by livecode-vulcan** [comment]
😎 test success 82309bddd853725a290eee631954fad97cad050e

* try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/737)
* try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/742)
* try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/735)
* try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/736)
* try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/742)
* try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/733)
* try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/734)

**Pull request merged**